File: practice/bronze/2020/Jan/Word-Processor/Word.java
/***********************************
 *
 * Problem: Word Processor
 * Date: 2020-9-27
 * Author: Terry Gu
 *
 ***********************************/

/**
 * The essay contains N words (1≤N≤100), separated by spaces. Each word is
 * between 1 and 15 characters long, inclusive, and consists only of uppercase
 * or lowercase letters. According to the instructions for the assignment, the
 * essay has to be formatted in a very specific way: each line should contain no
 * more than K (1≤K≤80) characters, not counting spaces. Fortunately, Bessie's
 * word processor can handle this requirement, using the following strategy:
 * 
 * If Bessie types a word, and that word can fit on the current line, put it on
 * that line. Otherwise, put the word on the next line and continue adding to
 * that line. Of course, consecutive words on the same line should still be
 * separated by a single space. There should be no space at the end of any line.
 * 
 * Unfortunately, Bessie's word processor just broke. Please help her format her
 * essay properly!
 * 
 * INPUT FORMAT (file word.in): The first line of input contains two
 * space-separated integers N and K. The next line contains N words separated by
 * single spaces. No word will ever be larger than K characters, the maximum
 * number of characters on a line.
 * 
 * OUTPUT FORMAT (file word.out): Bessie's essay formatted correctly.
 * 
 */

import java.io.*;
import java.util.*;

public class Word {

  public static void main(String[] args) throws IOException {

    BufferedReader br = new BufferedReader(new FileReader("word.in"));
    PrintWriter pw = new PrintWriter(new BufferedWriter(new FileWriter("word.out")));

    StringTokenizer st = new StringTokenizer(br.readLine());

    int n = Integer.parseInt(st.nextToken());
    int k = Integer.parseInt(st.nextToken());

    

    br.close();
    pw.close();

  }

}

File: USACO/bronze/2019-2020/Jan/Word-Processor/word.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;
using ll = long long;

ifstream fin("word.in");
ofstream fout("word.out");

int N, K;

int main() {

  fin >> N >> K;

  int current;
  for (ll i = 0; i < N; i++) {
    string word;
    fin >> word;

    if(current == 0) {
      fout << word;
      current += word.size();
    }

    else if (current + word.size() <= K) {
      fout << " " + word;
      current += word.size();
    } else if (current + word.size() > K) {
      fout << "\n" + word;
      current = word.size();
      // cout << "NEXT LINE" << endl;
    }

    // cout << word << " " << current << endl;
  }
}